ui/src/app/services/groups/group.service.ts

```diff
@@ -48,42 +48,76 @@ export class GroupService {
             .catch(this.utils.catchError);
     }
 
-    leaveGroup(userId, groupId) {
-        return this.utils.setTimeoutPromise(1000)
-            .then(() => {
-                return true;
+    addOwnerToGroup(groupId, member) {
+        let request = environment.add_owner(groupId, member);
+        return this.http.post(request.url, request.body, this.context.httpOptions())
+            .toPromise()
+            .then((response) => {
+                console.log('Add Owner: ',response.json());
+                return response.json();
+            })
+            .catch(this.utils.catchError);
+    }
+
+    addAdminToGroup(groupId, member) {
+        let request = environment.add_admin(groupId, member);
+        return this.http.post(request.url, request.body, this.context.httpOptions())
+            .toPromise()
+            .then((response) => {
+                console.log('Add Admin: ',response.json());
+                return response.json();
             })
             .catch(this.utils.catchError);
     }
 
+    leaveGroup(userId, groupId) {
+        return this.removeFromGroup(userId, groupId);
+    }
+
     promoteToOwner(userId, groupId) {
-        return this.utils.setTimeoutPromise(500)
-            .then(() => {
-                return true;
+        let request = environment.promote_member_to_role_owner(groupId, userId);
+        return this.http.patch(request.url, request.body, this.context.httpOptions())
+            .toPromise()
+            .then((response) => {
+                console.log('Promote member to owner: ',response.json());
+                return response.json();
             })
             .catch(this.utils.catchError);
     }
 
     promoteAllToOwner(selection) {
-        return this.utils.setTimeoutPromise(500)
-            .then(() => {
-                return true;
+        const groupId = selection[0].memberOf[0];
+        const member = selection[0].id;
+        let request = environment.promote_member_to_role_owner(groupId, member);
+        return this.http.patch(request.url, request.body, this.context.httpOptions())
+            .toPromise()
+            .then((response) => {
+                console.log('Promote member to owner: ',response.json());
```

Comment:
Should the message here be "Promote all to owner" or something that differentiates it from the log message in promoteToOwner()?

Reply:
the "promoteAllToOwner" method does not seem right. we will need to fix it in the other PR. this log message seems proper for this method. Will use a different log message once we are working to fix "promoteAllToOwner"

Reply:
Ok that works.